### ISSUE TITLE
[Block: 5927] - '/osmosis.gamm.v1beta1.MsgSwapExactAmountIn'.

### ISSUE THREAD

**Issue by pharr117** [DefiantLabs/cosmos-tax-cli#349]
# Summary 
This block fails on pre-validation through all 3 current MsgSwapExactAmountIn parsers due to the events not being in the expected form.

The solution will be to analyze the message for the alternate format of this message type and build an alternate parser for it.

## Deeper Analysis

The format for the events contains the AmountOut value in a later event in the event array, we need to get this value instead of what the other 3 parsers are attempting to get.

**Comment by pharr117** [comment]
I have already fixed this one in my branch.

I have added an alternate parser for this message type that takes into account the event format found for TX BB954377AB50F8EF204123DC8B101B7CB597153C0B8372166BC28ABDAA262516 in this block.

**Comment by pharr117** [comment]
This one keeps popping up because we are approaching the SwapExactAmountIn events in the wrong manner. We keep looking for specific events in the transfer event that fit the broken TXes as they keep popping up.

I have pushed a new commit to my branch that should take care of this going forward. The last element in the `transfer` event always contain the final transfer from the pool to the user based on their desired denom_out. So I have improved the parser from commit f7662392 by just looking for the final transfer event and validating.
